crates/modelardb_server/src/storage/uncompressed_data_manager.rs

```diff
@@ -93,61 +117,110 @@ impl UncompressedDataManager {
+                .unwrap();
+
+            // unwrap() is safe as data cannot be ingested into a model table that does not exist.
+            let model_table_metadata = context
+                .model_table_metadata_from_default_database_schema(&table_name)
+                .await
+                .unwrap()
+                .unwrap();
 
             for maybe_file_dir_entry in folder_dir_entry.path().read_dir()? {
-                finished_data_buffers.push_back(Box::new(UncompressedOnDiskDataBuffer::try_new(
+                let buffer = Box::new(UncompressedOnDiskDataBuffer::try_new(
                     univariate_id,
-                    error_bound,
+                    model_table_metadata.clone(),
                     maybe_file_dir_entry?.path(),
-                )?));
+                )?);
+
+                initial_disk_space += buffer.disk_size();
+
+                self.channels
+                    .univariate_data_sender
+                    .send(Message::Data(buffer))
+                    .map_err(|error| IOError::new(IOErrorKind::BrokenPipe, error))?;
             }
         }
 
         // Record the used disk space of the uncompressed data buffers currently on disk.
-        let initial_disk_space: usize = finished_data_buffers
-            .iter()
-            .map(|buffer| buffer.disk_size())
-            .sum();
-        used_disk_space_metric
-            .write()
-            .await
+        // unwrap() is safe as lock() only returns an error if the lock is poisoned.
+        self.used_disk_space_metric
+            .lock()
+            .unwrap()
             .append(initial_disk_space as isize, true);
 
-        Ok(Self {
-            local_data_folder,
-            current_batch_index: 0,
-            active_uncompressed_data_buffers: HashMap::new(),
-            finished_uncompressed_data_buffers: finished_data_buffers,
-            memory_pool,
-            compress_directly,
-            used_uncompressed_memory_metric: Metric::new(),
-            ingested_data_points_metric: Metric::new(),
-            used_disk_space_metric,
-        })
+        Ok(())
+    }
+
+    /// Read and process messages received from the [`super::StorageEngine`] to either ingest
+    /// uncompressed data, flush buffers, or stop.
+    pub(super) fn process_uncompressed_messages(
+        &self,
+        runtime: Arc<Runtime>,
+    ) -> Result<(), String> {
+        loop {
+            let message = self
+                .channels
+                .multivariate_data_receiver
+                .recv()
+                .map_err(|error| error.to_string())?;
+
+            match message {
+                Message::Data(uncompressed_data_multivariate) => {
+                    runtime
+                        .block_on(self.insert_data_points(uncompressed_data_multivariate))
+                        .map_err(|error| error.to_string())?;
+                }
+                Message::Flush => {
+                    self.flush().unwrap();
```

Comment:
They should have been removed, so I have added `flush_and_log_errors()` which calls `flush()` and logs any errors it may return. I decided not to just log errors in `flush()` so its signature makes it clear it can fail.